modules/programs/dircolors.nix

```diff
@@ -0,0 +1,221 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.dircolors;
+
+  formatLine = n: v: "${n} ${toString v}";
+in {
+  meta.maintainers = [ hm.maintainers.justinlovinger ];
+
+  options.programs.dircolors = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to manage <filename>.dir_colors</filename>
+        and set <code>LS_COLORS</code>.
+      '';
+    };
+
+    enableBashIntegration = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
+
+    enableFishIntegration = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to enable Fish integration.
+      '';
+    };
+
+    enableZshIntegration = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+
+    settings = mkOption {
+      type = with types; attrsOf str;
+      default = { };
+      description = ''
+        Options to add to <filename>.dir_colors</filename> file.
+        See <code>dircolors --print-database</code>
+        for options.
+      '';
+      example = literalExample ''
+        {
+          OTHER_WRITABLE = "30;46";
+          ".sh" = "01;32";
+          ".csh" = "01;32";
+        }
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Extra lines added to <filename>.dir_colors</filename> file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    programs.dircolors.settings = {
```

Comment:
From what I can tell these match the output of `dircolors -p`. Perhaps would be good to add a comment stating this?

Reply:
Done.